x-pack/legacy/plugins/task_manager/task_pool.ts

```diff
@@ -112,3 +121,23 @@ export class TaskPool {
     }
   }
 }
+
+function ifTaskManagerHasRanOutOfWorkerCapacity(callback: () => void) {
+  return function([result]: [TAKE_RESULT, TaskRunner]) {
+    if (result === TAKE_RESULT.RAN_OUT_OF_CAPACITY) {
+      callback();
+    }
+  };
+}
+
+function ifClaimingOwnershipThrewAnError(callback: () => void) {
+  return function([result]: [TAKE_RESULT, TaskRunner]) {
+    if (result === TAKE_RESULT.TAKE_FAILURE) {
+      callback();
+    }
+  };
+}
+
+function taskManagerHasClaimedOwnership([result, _]: [TAKE_RESULT, TaskRunner]) {
```

Comment:
I'm curious about the usage of `, _` in the `[result, _]` bit.  The `ifTaskManagerHasRanOutOfWorkerCapacity()` and `ifClaimingOwnershipThrewAnError()` functions above `taskManagerHasClaimedOwnership()` **don't** do this, but guessing the result is the same either way?

Reply:
Haha, it's a FP thing.

It's a way of denoting that _there's a variable there, but we don't use it, and so no point naming it_.
Come top think of it, as JS doesn't require you to specify vars you don't need and this isn't standard in JS, I'll just omit them where possible.

Reply:
And the reason taskManagerHasClaimedOwnership has it and the others don't is simply this: I'm human and I'm not always consistent. :)

The Rust/ML/Scala compilers *require* you to specify every variable, which means you can't omit args the way you can in JS, but it lets you name multiple tokens as *_* and you simply can't use them.
So it's a standard way of expressing that there is something there and it's meaningful, but we don't use it in the current scope so no need to add to cognitive load.

You might think _wait, but that adds noise which is cognitive load_, and that's true, but it also prevents a whole range of mistakes that can happen when you don't have this requirement, for example:
```javascript
const val = ["1.2","2.5","3.7"].map(parseInt); // you expect to get back [1, 2, 3]
console.log(val); // but you get [1, NaN, NaN]
```

It's because in JS you're not required to address all args and so you forget about args (in this case `index` which gets passed into `base`), but in Scala/Rust/ML, you'd be forced to do this:
```javascript
const val = ["1.2","2.5","3.7"].map((val, _) => parseInt(val)); 
console.log(val); // you get [1, 2, 3]
```
Which keeps you from making such mistakes. :)
Anyway, without the compiler enforcing this, there isn't much point using the `_` and in fact, it's confusing if you don't know what the intention is, so I'll omit it.

Reply:
Another reason not to use _ the way I have, is that in Javascript we can do something that you can't do in the above mentioned languages: `map(([, task]) => task)`.
I can actually omit the former index when destructuring, which means you don't need the *_* in `map(([_, task]) => task)`.

In the other languages, that is a syntax error.
I personally prefer compilers to be more terse than this and would prefer to simply have *_*s, but seeing as it's not going to be clear to newcomers and isn't idiomatic to JS, I'm happy to omit these.